vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/EagerApplicationScopedExtension.java

```diff
@@ -0,0 +1,38 @@
+package br.com.caelum.vraptor.ioc.cdi.extension;
+
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.util.AnnotationLiteral;
+
+import br.com.caelum.vraptor.ioc.cdi.EagerInitialized;
+
+/**
+ * When enabled, it allows users to use {@link EagerInitialized} annotation 
+ * to get your application scoped beans initialized just after CDI deployment.
+ *
+ * @author Rodrigo Turini
+ */
+public class EagerApplicationScopedExtension implements Extension {
+	
+	@SuppressWarnings("serial")
+	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
+		
+		Set<Bean<?>> beans = beanManager.getBeans(Object.class, 
+				new AnnotationLiteral<EagerInitialized>() {});
+		
+		for (Bean<?> bean : beans) {
+			Class<?> beanClass = bean.getBeanClass();
+			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
+				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
+				beanManager.getReference(bean, beanClass, ctx).toString();
```

Comment:
Besides that, I like the idea :).

Em ter, 3 de fev de 2015 23:43, Alberto SOUZA <alots.ssa@gmail.com>
escreveu:

> I think we should use the @Startup annotation. It has a well defined
> meaning... Just add the ejb-spec jar and it is done. I don't think it is a
> good practice to create annotations that already exists.
>
> Em ter, 3 de fev de 2015 20:26, Lucas Cavalcanti <notifications@github.com>
> escreveu:
>
> In vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/
>> EagerApplicationScopedExtension.java
>> <https://github.com/caelum/vraptor4/pull/937#discussion_r24046038>:
>>
>> > + *
>> > + * @author Rodrigo Turini
>> > + */
>> > +public class EagerApplicationScopedExtension implements Extension {
>> > +	
>> > +	@SuppressWarnings("serial")
>> > +	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
>> > +		
>> > +		Set<Bean<?>> beans = beanManager.getBeans(Object.class,
>> > +				new AnnotationLiteral<EagerInitialized>() {});
>> > +		
>> > +		for (Bean<?> bean : beans) {
>> > +			Class<?> beanClass = bean.getBeanClass();
>> > +			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
>> > +				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
>> > +				beanManager.getReference(bean, beanClass, ctx).toString();
>>
>> don't we have a cdi method to initialize the class? toString on arbitrary
>> classes is not exactly good. #gambiarra
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/caelum/vraptor4/pull/937/files#r24046038>.
>>
>